### ISSUE TITLE
proposed change to loadPixels in p5.MediaElement

### ISSUE THREAD

**Issue by shiffman** [processing/p5.js#1235]
I am working on an example that save a video image each frame for motion detection.  Right now I have to write the code as follows:

```javascript
function draw() {
  if (video.canvas) {
    prevFrame.copy(video, 0, 0, 320, 240, 0, 0, 320, 240);
  }
}
```

I have to check if the canvas has been created b/c for the first few frames the video hasn't actually started.  This is due to how `loadPixels()` is written inside `p5.MediaElement` where a canvas is not made unless `this.loadedmetadata` is `true`.  If I rewrite that function as such:

```javascript
  p5.MediaElement.prototype.loadPixels = function() {
    // *** Now the canvas is always created ****
    if (!this.canvas) {
      this.canvas = document.createElement('canvas');
      this.drawingContext = this.canvas.getContext('2d');
    }
    if (this.loadedmetadata) { // wait for metadata for w/h
      if (this.canvas.width !== this.elt.width) {
        this.canvas.width = this.elt.width;
        this.canvas.height = this.elt.height;
        this.width = this.canvas.width;
        this.height = this.canvas.height;
      }
      this.drawingContext.drawImage(this.elt, 0, 0, this.canvas.width, this.canvas.height);
      p5.Renderer2D.prototype.loadPixels.call(this);
    }
    return this;
  }
```

I can then more simply say in my code:

```javascript
function draw() {
  prevFrame.copy(video, 0, 0, 320, 240, 0, 0, 320, 240);
}
```

@lmccart shall I implement this as a pull request or will this cause other issues?

**Comment by lmccart** [comment]
i think this looks good!

**Comment by lmccart** [comment]
closed with #1236
